### PR TITLE
Update numfmt.jf and currency locale data to CLDR 36 

### PR DESCRIPTION
**Pull request by gouniLee**
### Checklist

* [ ] At least one test case is included for this feature or bug fix.
* [ ] `ReleaseNotes` has added or is not needed.
* [x] This PR has passed all of the test cases on `Nodejs` and `Chrome Browser`.
* [ ] This PR has passed all of the test cases on `QT/QML`.
* [ ] This is an API breaking change.
* [x] Requires a major version change.

### Issue Resolved / Feature Added
Update `currency.json`, `currency.jf` and `numfmt.jf` locale data to CLDR 36 by running `gennumfmt.js` and `gencurrencies.js`. Fixed all test case failures. 

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)

### Links
[//]: # (Related issues, references)

### CODE REVIEW

File: js/data/locale/currency.jf [hunk 1]
```diff
@@ -1 +0,0 @@
-{"currency": "USD"}
```
Comment (ehoogerbeets) [inline js/data/locale/currency.jf]: The root currency should be US Dollars if no others are specified

File: js/data/locale/currency.json [hunk 2]
```diff
@@ -32,7 +32,7 @@
     "KES": {
         "name": "Kenyan Shilling",
         "decimals": 2,
-        "sign": "Sh"
+        "sign": "KES"
```
Comment (ehoogerbeets) [inline js/data/locale/currency.json]: The sign should contain the common use symbols used in the country for that currency, not the ISO code. There is another option on the number formatter to use the ISO code instead.

File: js/data/locale/currency.json [hunk 3]
```diff
@@ -42,32 +42,32 @@
     "CHF": {
         "name": "Swiss Franc",
         "decimals": 2,
-        "sign": "Fr"
+        "sign": "CHF"
```
Comment (ehoogerbeets) [inline js/data/locale/currency.json]: The symbols are in the "symbol-alt-narrow" property of the cldr json files.

File: js/data/locale/currency.jf (hunk 1 above)
Comment (gouniLee) [reply js/data/locale/currency.jf]: I'll rollback the file. I'll update the extractor tool code so that it can be automatically generated from the tool code.

File: js/data/locale/currency.json (hunk 3 above)
Comment (gouniLee) [reply js/data/locale/currency.json]: I'll update the extractor code then regenerate files.

File: js/test/number/testnumfmt.js [hunk 4]
```diff
@@ -4073,7 +4073,7 @@ module.exports.testnumfmt = {
     
         test.ok(fmt);
     
-        test.equal(fmt.format(100110.57), "ƒ 100.110,57");
```
Comment (ehoogerbeets) [inline js/test/number/testnumfmt.js]: Most of these changes are changing from double quote to single quote, which does not make any semantic difference. I would not make these sort of unnecessary changes in a PR that is mixed with actual code changes from CLDR because it hides the real changes. Things like fixing whitespace or changing quotes or other such "clean up" type of changes should go into separate PRs.

Comment (gouniLee) [reply js/test/number/testnumfmt.js]: Yeah, I think I'd better closing this PR, then re-create PR to see changes clearly and I'll separate clean up PR if it's necessary.
